vignettes/intro.Rmd

```diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+             )
+           ),
+           card(class = "blue",
+             div(class = "content",
+                 div(class = "header", "Main title card 2"),
+                 div(class = "meta", "Sub title card 2"),
+                 div(class = "description", "More detail description card 2")
+                 )
+             )
+           )
+         )
+      )
+    )
+  )
+
+server <- function(input, output, session) {}
+
+shinyApp(ui, server)
+```
+
+Time to fill in sidebar `sidebar_panel()`, so let's add dropdown. We create a dropdown input using `dropdown_input()`.
+
+```{r}
+ui <- semanticPage(
+  title = "My first page",
+  h1("My page"),
+  sidebar_layout(
+    sidebar_panel(
+      uiOutput("dropdown")
+    ),
+    main_panel(
+      segment(
+         cards(
+           class = "two",
+           card(class = "red",
+             div(class = "content",
+                 div(class = "header", "Main title card 1"),
+                 div(class = "meta", "Sub title card 1"),
+                 div(class = "description", "More detail description card 1")
+             )
+           ),
+           card(class = "blue",
+             div(class = "content",
+                 div(class = "header", "Main title card 2"),
+                 div(class = "meta", "Sub title card 2"),
+                 div(class = "description", "More detail description card 2")
+             )
+           )
+         )
+      )
+      )
+    )
+)
+
+server <- function(input, output, session) {
+  output$dropdown <- renderUI({
+    dropdown_input("mtcars_dropdown", c("mpg", "cyl", "disp", "hp"), value = "mpg")
+  })
+}
+
+shinyApp(ui, server)
+```
+
+Let's make dropdown do something and generate a histogram from a selected `mtcars` column and a plot.
+
+```{r}
+ui <- semanticPage(
+  title = "My first page",
+  h1("My page"),
+  sidebar_layout(
+    sidebar_panel(
+      p("Select variable for plots:"),
+      uiOutput("dropdown")
+    ),
+    main_panel(
+      segment(
+         cards(
+           class = "two",
+           card(class = "red",
+             div(class = "content",
+                 div(class = "header", "Main title card 1"),
+                 div(class = "meta", "Sub title card 1"),
+                 div(class = "description", "More detail description card 1")
+             )
+           ),
+           card(class = "blue",
+             div(class = "content",
+                 div(class = "header", "Main title card 2"),
+                 div(class = "meta", "Sub title card 2"),
+                 div(class = "description", "More detail description card 2")
+             )
+           )
+         )
+      ),
+      plotOutput("plot"),
+      plotOutput("plot2")
+      )
+    )
+)
+
+server <- function(input, output, session) {
+  output$dropdown <- renderUI({
+    dropdown_input("mtcars_dropdown", c("mpg", "cyl", "disp", "hp"), value = "mpg")
+  })
+
+  data_plot <- reactive({
```

Comment:
is this really necessary? it might create some confusion, can we just render plot straight from `mtcars[[input$mtcars_dropdown]]`

Reply:
alternatively explain this step in details

Reply:
done